src/Form/Type/UserGenderListType.php

```diff
@@ -15,6 +15,17 @@
 
 use Sonata\CoreBundle\Form\Type\BaseStatusType;
 
+@trigger_error(
+    'The '.__NAMESPACE__.'\UserGenderListType class is deprecated since version 4.x and will be removed in 5.0.'
+    .' Use choice type instead.',
+    E_USER_DEPRECATED
+);
+
+/**
+ * NEXT_MAJOR: remove this class.
+ *
+ * @deprecated since version 4.x, to be removed in 5.0. Use choice type instead
```

Comment:
Why did you deprecate this?

Couldn't you move the stuff from the `UserAdmin` right here?

Reply:
why would I use a class to solve this simple choice thing?

Reply:
The problem is, that the `UserGenderListType` is broken, not the admin class (only)

Reply:
Is broken and cannot be fixed unless we allow deprecations (adding the fourth parameter to false) or more BC breaks. And it is because of this other PR: https://github.com/sonata-project/SonataCoreBundle/pull/499 which breaks BC.

My first thought was to implement a new AbstractStatusType but I there are 2 reasons why I didn't do it:

1. I would need to change which class is extending UserGenderListType (probably BC break too)
2. This can be solved in a much easier way, without using this class, and using a simple choiceType.

How to proceed? At this moment the user admin class is unusable

Reply:
Can't you just add this to the form type?

```
        // NEXT_MAJOR: Remove this when dropping support for SF 2.8
        if (method_exists(FormTypeInterface::class, 'setDefaultOptions')) {
            $genderOptions['choices_as_values'] = true;
        }
```